pkg/azurecluster/update.go

```diff
@@ -0,0 +1,60 @@
+package azurecluster
+
+import (
+	"context"
+	"strings"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/label"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/validator"
+)
+
+type UpdateValidator struct {
+	logger micrologger.Logger
+}
+
+type UpdateValidatorConfig struct {
+	Logger micrologger.Logger
+}
+
+func NewUpdateValidator(config UpdateValidatorConfig) (*UpdateValidator, error) {
+	v := &UpdateValidator{
+		logger: config.Logger,
+	}
+
+	return v, nil
+}
+
+func (a *UpdateValidator) Validate(ctx context.Context, request *v1beta1.AdmissionRequest) (bool, error) {
+	azureClusterNewCR := &capzv1alpha3.AzureCluster{}
+	azureClusterOldCR := &capzv1alpha3.AzureCluster{}
+	if _, _, err := validator.Deserializer.Decode(request.Object.Raw, nil, azureClusterNewCR); err != nil {
+		return false, microerror.Maskf(parsingFailedError, "unable to parse AzureCluster CR: %v", err)
+	}
+	if _, _, err := validator.Deserializer.Decode(request.OldObject.Raw, nil, azureClusterOldCR); err != nil {
+		return false, microerror.Maskf(parsingFailedError, "unable to parse AzureCluster CR: %v", err)
+	}
+
+	oldClusterVersion := azureClusterOldCR.Labels[label.ReleaseVersion]
```

Comment:
The specific logic of validating release labels of two objects should probably be extracted into own package and used in all validators. That way it's easier to change later if we want (and we might want). WDYT? It could accept [metav1.Object](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#Object) interface as param and do the thing from there.

Reply:
Oh, yeah, another thought: This way you could easily write test cases without too much boilerplate or setup. :slightly_smiling_face:

Reply:
Yes I am trying to remove some code duplication with a similar approach. Let's see how it turns out